ecadlabs/taqueria
# Build SDK using tsup

# 🏗️ PR ➾

Fixes # 1229

## 🪂 Pre-Merge Checklist

- [x] 🦀 Automated tests have been written and added to this PR
- [x] 🤿 Corresponding changes have been made to all documentation
- [x] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] 🏄‍♂️ Myself and others have built this branch and done manual testing on the change
- [ ] ⛵ A summary describing the change has been written for release notes

----------------------------------------------------------------------------------------------------------------------------

## 🪁 Description

Adjusted SDK to use tsup, which allows us to build typescript declaration files. This is the first step towards standardizing on tsup as our build tool for this repo.

The Dappetizer team discovered that we were doing the following, all of which this PR should address:
- We're including *.ts source files in our SDK npm package
- We're not generating *.d.ts files for our SDK
- We're including tsconfig.json in our npm package

## 🎢 Test Plan

1. Checkout this branch
2. Run the following from the root of your git clone for this repo:

```shell
npm pack -w taqueria-sdk
npm pack -w taqueria-protocol
```

3. Run `tar -ztvf taqueria*.tgz` and verify that no files are included that shouldn't be. 

> NOTE: I also added an integration-test which performs the above.

----------------------------------------------------------------------------------------------------------------------------

### 🛸 Type of Change

- [x] 🐟 Minor change (non-breaking change of very limited scope)
- [ ] 🐳 Major refactor (breaking changes and significant impact to the codebase)

----------------------------------------------------------------------------------------------------------------------------
